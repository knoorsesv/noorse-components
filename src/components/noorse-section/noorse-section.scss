noorse-section {

  display: inline-block;
  width: fit-content;
  padding: 20px;

  h1 {
    display: inline-block;
    border-bottom: 2px solid #008000;
  }

  h2 {
    font-weight: bold;
  }
}
