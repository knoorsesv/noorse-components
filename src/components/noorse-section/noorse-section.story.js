import {console, document} from 'global'
import {storiesOf} from '@storybook/html'

storiesOf('Component example', module)
  .add('without content', () => `<noorse-component></noorse-component>`)
