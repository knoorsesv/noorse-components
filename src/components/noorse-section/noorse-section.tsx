import {Component, h, Prop} from '@stencil/core';

@Component({
  tag: 'noorse-section',
  styleUrl: 'noorse-section.scss'
})
export class NoorseSection {

  @Prop() header: string;


  render() {
    return <div>
      {this.header ? <h1>{this.header}</h1> : ''}
      <p>
        <slot/>
      </p>
    </div>;
  }
}
