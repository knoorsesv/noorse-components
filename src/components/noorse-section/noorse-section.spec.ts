import {newSpecPage} from "@stencil/core/testing";
import {SpecPage} from "@stencil/core/dist/declarations";
import {NoorseSection} from "./noorse-section";

describe.only("Noorse Section", function () {

  let card: SpecPage;


  it('with header', async () => {
    card = await newSpecPage({
      components: [NoorseSection],
      html: `<noorse-section header="header">content</noorse-section>`
    });
    await card.waitForChanges()
    expect(card.root.outerHTML).toMatchSnapshot()
  });

  it('without header', async () => {
    card = await newSpecPage({
      components: [NoorseSection],
      html: `<noorse-section>content</noorse-section>`
    });
    await card.waitForChanges()
    expect(card.root.outerHTML).toMatchSnapshot()
  });

});
