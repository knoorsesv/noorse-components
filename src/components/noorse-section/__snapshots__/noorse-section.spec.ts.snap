// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Noorse Section with header 1`] = `"<noorse-section header=\\"header\\"><!----> <div><h1>header</h1><p> content</p></div></noorse-section>"`;

exports[`Noorse Section without header 1`] = `"<noorse-section><!----> <div> <p> content</p></div></noorse-section>"`;
